frame/staking/src/lib.rs

```diff
@@ -2299,11 +2299,21 @@ impl<T: Trait> Module<T> {
 		Self::bonded(stash).and_then(Self::ledger).map(|l| l.active).unwrap_or_default()
 	}
 
-	/// internal impl of [`slashable_balance_of`] that returns [`VoteWeight`].
-	fn slashable_balance_of_vote_weight(stash: &T::AccountId) -> VoteWeight {
-		<T::CurrencyToVote as Convert<BalanceOf<T>, VoteWeight>>::convert(
-			Self::slashable_balance_of(stash)
-		)
+	/// Internal impl of [`slashable_balance_of`] that returns [`VoteWeight`].
+	fn slashable_balance_of_vote_weight(stash: &T::AccountId, issuance: BalanceOf<T>) -> VoteWeight {
+		T::CurrencyToVote::to_vote(Self::slashable_balance_of(stash), issuance)
+	}
+
+	/// Returns a closure around `slashable_balance_of_vote_weight` that can be passed around.
+	///
+	/// This prevents call sites from repeatedly requesting `total_issuance` from backend.
+	pub(crate) fn weight_of_fn() -> Box<dyn Fn(&T::AccountId) -> VoteWeight> {
+		// TODO: changing this to unboxed `impl Fn(..)` return type and the module will still
```

Comment:
This would be interesting for anyone reviewing to try out.

Reply:
Have you tried forcing the lifetime to `'static` (as `Box` do under the hood)?

Reply:
Made some attempts, ended up with the same error message: 

```log
rror[E0277]: expected a `std::ops::Fn<(&u64,)>` closure, found `impl for<'r> std::ops::Fn<(&'r <mock::Test as frame_system::Trait>::AccountId,)>`
   --> frame/staking/src/mock.rs:902:3
    |
902 |         Staking::weight_of_fn(),
    |         ^^^^^^^^^^^^^^^^^^^^^^^ expected an `Fn<(&u64,)>` closure, found `impl for<'r> std::ops::Fn<(&'r <mock::Test as frame_system::Trait>::AccountId,)>`
    |
   ::: /home/kianenigma/remote-builds/932196901925798239/primitives/npos-elections/src/helpers.rs:29:20
    |
29  |     stake_of: impl Fn(&A) -> VoteWeight + 'static,
    |                    -------------------- required by this bound in `_npos::assignment_ratio_to_staked`
    |
    = help: the trait `for<'r> std::ops::Fn<(&'r u64,)>` is not implemented for `impl for<'r> std::ops::Fn<(&'r <mock::Test as frame_system::Trait>::AccountId,)>`

error[E0277]: expected a `std::ops::FnOnce<(&u64,)>` closure, found `impl for<'r> std::ops::Fn<(&'r <mock::Test as frame_system::Trait>::AccountId,)>`
   --> frame/staking/src/mock.rs:902:3
    |
902 |         Staking::weight_of_fn(),
    |         ^^^^^^^^^^^^^^^^^^^^^^^ expected an `FnOnce<(&u64,)>` closure, found `impl for<'r> std::ops::Fn<(&'r <mock::Test as frame_system::Trait>::AccountId,)>`
    |
   ::: /home/kianenigma/remote-builds/932196901925798239/primitives/npos-elections/src/helpers.rs:29:30
    |
29  |     stake_of: impl Fn(&A) -> VoteWeight + 'static,
    |                              ---------- required by this bound in `_npos::assignment_ratio_to_staked`
    |
    = help: the trait `std::ops::FnOnce<(&u64,)>` is not implemented for `impl for<'r> std::ops::Fn<(&'r <mock::Test as frame_system::Trait>::AccountId,)>`
```